diff --git a/src/main/java/edu/illinois/library/cantaloupe/image/exif/Reader.java b/src/main/java/edu/illinois/library/cantaloupe/image/exif/Reader.java
index bc4261823..1c4ba8037 100644
--- a/src/main/java/edu/illinois/library/cantaloupe/image/exif/Reader.java
+++ b/src/main/java/edu/illinois/library/cantaloupe/image/exif/Reader.java
@@ -155,10 +155,7 @@ public final class Reader implements AutoCloseable {
 
     private byte[] readBytes(int length) throws IOException {
         byte[] data = new byte[length];
-        int n, offset = 0;
-        while ((n = inputStream.read(data, offset, data.length - offset)) < offset) {
-            offset += n;
-        }
+        inputStream.readFully(data);
         return data;
     }
 
diff --git a/src/main/java/edu/illinois/library/cantaloupe/processor/codec/gif/GIFMetadataReader.java b/src/main/java/edu/illinois/library/cantaloupe/processor/codec/gif/GIFMetadataReader.java
index 1978807d2..14887dad3 100644
--- a/src/main/java/edu/illinois/library/cantaloupe/processor/codec/gif/GIFMetadataReader.java
+++ b/src/main/java/edu/illinois/library/cantaloupe/processor/codec/gif/GIFMetadataReader.java
@@ -340,11 +340,7 @@ final class GIFMetadataReader implements AutoCloseable {
 
     private byte[] read(int length) throws IOException {
         byte[] data = new byte[length];
-        int n, offset = 0;
-        while ((n = inputStream.read(
-                data, offset, data.length - offset)) < offset) {
-            offset += n;
-        }
+        inputStream.readFully(data);
         return data;
     }
 
diff --git a/src/main/java/edu/illinois/library/cantaloupe/processor/codec/jpeg/JPEGMetadataReader.java b/src/main/java/edu/illinois/library/cantaloupe/processor/codec/jpeg/JPEGMetadataReader.java
index b54056c0c..44be18b2d 100644
--- a/src/main/java/edu/illinois/library/cantaloupe/processor/codec/jpeg/JPEGMetadataReader.java
+++ b/src/main/java/edu/illinois/library/cantaloupe/processor/codec/jpeg/JPEGMetadataReader.java
@@ -296,11 +296,7 @@ public final class JPEGMetadataReader {
 
     private byte[] read(int length) throws IOException {
         byte[] data = new byte[length];
-        int n, offset = 0;
-        while ((n = inputStream.read(
-                data, offset, data.length - offset)) < offset) {
-            offset += n;
-        }
+        inputStream.readFully(data);
         return data;
     }
 
diff --git a/src/main/java/edu/illinois/library/cantaloupe/processor/codec/jpeg2000/JPEG2000MetadataReader.java b/src/main/java/edu/illinois/library/cantaloupe/processor/codec/jpeg2000/JPEG2000MetadataReader.java
index ab9252d2a..24da4bef5 100644
--- a/src/main/java/edu/illinois/library/cantaloupe/processor/codec/jpeg2000/JPEG2000MetadataReader.java
+++ b/src/main/java/edu/illinois/library/cantaloupe/processor/codec/jpeg2000/JPEG2000MetadataReader.java
@@ -434,11 +434,7 @@ public final class JPEG2000MetadataReader implements AutoCloseable {
 
     private byte[] read(int length) throws IOException {
         byte[] data = new byte[length];
-        int n, offset = 0;
-        while ((n = inputStream.read(
-                data, offset, data.length - offset)) < offset) {
-            offset += n;
-        }
+        inputStream.readFully(data);
         return data;
     }
 
